crates/wiggle/src/guest_type.rs

```diff
@@ -88,7 +88,7 @@ macro_rules! primitives {
                 if ptr.mem().is_borrowed(region) {
                     return Err(GuestError::PtrBorrowed(region));
                 }
-                Ok(unsafe { *host_ptr.cast::<Self>() })
+                Ok(unsafe { <$i>::from_le_bytes(*host_ptr.cast::<[u8; mem::size_of::<Self>()]>()) })
```

Comment:
Instead of using `[u8; ...]` could the `from_le` and `to_le` methods be used?

Reply:
I think when I tried those, they failed for f32 and f64 types.

Reply:
Ah yeah that'd do it!